File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        cost.sort()
        j = len(cost) - 1
        ans = 0
        while(j > 0):
            for i in range(2):
                ans += cost[j]
                j-=1
            
            j-=1
        
        if j==0:
            ans+=cost[j]
        return ans

File: 2144-minimum-cost-of-buying-candies-with-discount/2144-minimum-cost-of-buying-candies-with-discount.java
class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);
        int j = cost.length - 1;
        int ans = 0;
        while(j > 0){
            for(int i = 0;i < 2;i++){
                ans += cost[j];
                j--;
            }
            j--;
        }
        if(j==0)ans+=cost[j];
        return ans;
    }
}